### DISCUSSION TITLE
Andrew Abraham - Trend Trader Strategy

### DISCUSSION THREAD

**Discussion by M22arius494** [DaveSkender/Stock.Indicators#926] [Ideas]
Hello Dave,

I wanted to incorporate this in a scanner. Its the Trend Trader Strategy of Andrew Abraham. He wrote an article about it in "Trading the Trend" TASC September 1998, http://traders.com/Documentation/FEEDbk_docs/1998/09/Abstracts_new/Abraham/Abraham9809.html
The simple description of this is "The calculation uses a 21-period weighted average of the true range, giving higher weight to the true range of the most recent bar. The final value is then multiplied by 3 (and lot more, but this what I struggle with)."

There exists a Pine script for it (https://in.tradingview.com/script/j1etwXMQ-Trend-Trader-Strategy/ its marked as open source). Now I'm trying to understand what this script is doing but I seem to fail quite miserably at the translation of the line "avgTR = ta.wma(ta.atr(1), 21)"

From what I understand (from the description) it should calculate the ATR values (of the last 21 candles) and with that calculate the WMA from it, but I got weird values. How can I get the last 21 values from the ATR into the WMA?

Greetings,
Marius

**Comment by DaveSkender** [comment]
That `avgTR` piece that you're showing from TV is equivalent to doing this in our library:

```csharp
var results = quotes
   .GetTr()
   .GetWma(21);
```
And I agree, that "final value is then multiplied by 3" is confusing.  It's possible that of the 21 periods in the lookback window, they weight the last period 3×TR or 3× the normal weight when calculating the average of TR.  Not sure.

Even with figuring that out, TR and price are not on the same scale and the description of how to map it to price scale to get it to look like a stop-and-reverse is also confusing.  More research needed.

**Comment by M22arius494** [comment]
Aha, I see my error now, the article talks about the "True Range" (not "Average True Range") which then get averaged by the WMA. And there is a difference with the pine script (which uses ta.atr). Not sure where that difference came from but I should have picked that up sooner!

I'm also not exactly sure how this works, I'm just following up on some interesting strategies I picked up this year. I want to test this [Trend Trader Strategy](https://www.youtube.com/watch?v=2vn5C45lMVs) via an emulator and if that gives interesting results I want to manual test trade it a couple of times and from there we will see where it ends (most of those youtube stuff ends up in the garbage anyway).

Thanks Dave!

**Reply by DaveSkender** [reply]
yah, ATR(1) is a bit of a funny hack (an average of a single TR value) to reverse compute back to TR

**Reply by DaveSkender** [reply]
Just thinking through the commutative and distributive properties of math on this one, I bet you can algebraically take it from a long-winding 10-step process back to a 3-4 step process, and then it ends up looking nearly exactly like Wilder's Volatility Stop or ATR Trailing Stop, but just using WMA instead of SMA.
